### PR TITLE
[Security][SecurityBundle] Use csrf_token_id instead of deprecated intention

### PR DESCRIPTION
**Pull request by jakzal**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #16720 |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

File: src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php [hunk 1]
```diff
@@ -299,7 +299,7 @@ private function createFirewall(ContainerBuilder $container, $id, $firewall, &$a
             $listener = $container->setDefinition($listenerId, new DefinitionDecorator('security.logout_listener'));
             $listener->replaceArgument(3, array(
                 'csrf_parameter' => $firewall['logout']['csrf_parameter'],
-                'intention' => $firewall['logout']['csrf_token_id'],
+                'csrf_token_id' => $firewall['logout']['csrf_token_id'],
```
Comment (xabbuh) [inline src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php]: I think this is wrong as the `LogoutListener` doesn't handle an `intention` option (or we need to change the listener too which probably is better): https://github.com/symfony/symfony/blob/2.8/src/Symfony/Component/Security/Http/Firewall/LogoutListener.php#L62-L66

Comment (jakzal) [inline src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php]: I'm making changes to the security component just now :)

Comment (xabbuh) [inline src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php]: :+1:

**Comment by jakzal** [comment]
@xabbuh thanks for the review!

@nicolas-grekas there are some common changes made here and in the already merged #16692. Expect merge conflicts ;)

**Comment by jakzal** [comment]
I cleaned up the commits now. Ready to review.

File: src/Symfony/Bundle/SecurityBundle/CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,7 @@ CHANGELOG
 
  * deprecated the `key` setting of `anonymous`, `remember_me` and `http_digest`
    in favor of the `secret` setting.
+ * deprecated the `intention` firewall listener setting in favor of the `csrf_token_id`.
```
Comment (xabbuh) [inline src/Symfony/Bundle/SecurityBundle/CHANGELOG.md]: Is this change also documented in the upgrade file?

File: src/Symfony/Component/Security/Http/Firewall/LogoutListener.php [hunk 3]
```diff
@@ -57,11 +57,17 @@ public function __construct(TokenStorageInterface $tokenStorage, HttpUtils $http
             throw new InvalidArgumentException('The CSRF token manager should be an instance of CsrfProviderInterface or CsrfTokenManagerInterface.');
         }
 
+        if (isset($options['intention'])) {
+            @trigger_error('The "intention" option for the '.__CLASS__.' is deprecated since version 2.8 and will be removed in 3.0. Use the "csrf_token_id" option instead.', E_USER_DEPRECATED);
+
+            $options['csrf_token_id'] = $options['intention'];
+        }
```
Comment (xabbuh) [inline src/Symfony/Component/Security/Http/Firewall/LogoutListener.php]: What about throwing an exception if both `intention` and `csrf_token_id` are set (like we do it with deprecated options in the different `Configuration` classes)?

**Comment by nicolas-grekas** [comment]
:+1: thanks!

File: src/Symfony/Component/Security/Http/Firewall/LogoutListener.php (hunk 3 above)
Comment (xabbuh) [inline src/Symfony/Component/Security/Http/Firewall/LogoutListener.php]: same for the other listeners

Comment (jakzal) [inline src/Symfony/Component/Security/Http/Firewall/LogoutListener.php]: Not sure if it's worth it in this case?

File: src/Symfony/Bundle/SecurityBundle/CHANGELOG.md (hunk 2 above)
Comment (jakzal) [inline src/Symfony/Bundle/SecurityBundle/CHANGELOG.md]: Now it is ;)

File: src/Symfony/Component/Security/Http/Firewall/LogoutListener.php (hunk 3 above)
Comment (nicolas-grekas) [inline src/Symfony/Component/Security/Http/Firewall/LogoutListener.php]: mistakes happen and we need to help people fix them so if it's not too much work I'd say yes...

Comment (sstok) [inline src/Symfony/Component/Security/Http/Firewall/LogoutListener.php]: :+1: for throwing an exception.

Comment (jakzal) [inline src/Symfony/Component/Security/Http/Firewall/LogoutListener.php]: Done.

**Comment by xabbuh** [comment]
:+1: 

Status: Reviewed

**Comment by fabpot** [comment]
Thank you @jakzal.

**Pull request closed without merging**